src/git.rs

```diff
@@ -252,12 +252,18 @@ impl Repository {
 
     pub fn reset_head(&self) -> Result<(), PerformError> {
         let mut origin = self.repository.find_remote("origin")?;
+        let original_head = self.head_oid()?;
         origin.fetch(
             &["refs/heads/*:refs/heads/*"],
             Some(&mut Self::fetch_options(&self.credentials)),
             None,
         )?;
         let head = self.head_oid()?;
+
+        if head != original_head {
+            println!("Resetting index from {} to {}", original_head, head);
+        }
```

Comment:
maybe I'm misunderstanding the logic here, but do we even need to call `self.repository.reset()` if `head == original_head`?

Reply:
I avoided doing an early return here mainly to avoid any unexpected behavior changes to edge cases. It is possible that a previously failed job has staged changes in the index, but failed before the staged changes were committed. In that case, the local and remote `HEAD` commits would match, but the staged changes would become mingled with new changes from the next job.

I now actually wonder if a reset is always sufficient. I believe a reset will leave any untracked files alone. So it might be possible that the last job created the local file but failed before the new file was staged. In both cases I think these scenarios are extremely unlikely - we're much more likely to fail when accessing the network than for local file I/O - but maybe we should consider the equivalent of a `git clean -fdx` as well.

Reply:
if we're not sure about these edge cases, would it make sense to unconditionally log the operation?

Reply:
We could probably do the equivalent of `git status` and log a message if either the index or working directory isn't clean. Then we could log that as well if there is stale state hanging around. But for this message, I think it makes sense to only log it if the hashes are different.

Reply:
okay, sounds good to me